packages/circuit-ui/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,275 @@
+      if (preventClose) {
+        event.preventDefault();
+      }
+    }
+
+    useEffect(() => {
+      const dialogElement = dialogRef.current;
+      if (!dialogElement) {
+        return undefined;
+      }
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore The package is bundled incorrectly
+      dialogPolyfill.registerDialog(dialogElement);
+      if (preventClose) {
+        dialogElement.addEventListener('keydown', onPolyfillDialogKeydown);
+      }
+      if (onClose) {
+        dialogElement.addEventListener('close', onClose);
+      }
+
+      return () => {
+        if (onClose) {
+          dialogElement.removeEventListener('close', onClose);
+        }
+        if (!hasNativeDialog && dialogElement.nextSibling) {
+          (dialogElement.nextSibling as HTMLDivElement).removeEventListener(
+            'click',
+            onPolyfillBackdropClick,
+          );
+          dialogElement.removeEventListener('keydown', onPolyfillDialogKeydown);
+        }
+      };
+    }, [onClose]);
+
+    useEffect(() => {
+      const dialogElement = dialogRef.current;
+
+      if (!dialogElement) {
+        return undefined;
+      }
+      if (open) {
+        if (document.activeElement instanceof HTMLElement) {
+          lastFocusedElementRef.current = document.activeElement;
+        }
+        if (!dialogElement.open) {
+          dialogElement.showModal();
+          if (!hasNativeDialog) {
+            // use the polyfill backdrop
+            (dialogElement.nextSibling as HTMLDivElement).classList.add(
+              classes['backdrop-visible'],
+              classes.backdrop,
+            );
+            // intercept and prevent modal closing if preventClose is true
+            (dialogElement.nextSibling as HTMLDivElement).addEventListener(
+              'click',
+              onPolyfillBackdropClick,
+            );
+          }
+
+          // trigger show animation
+          dialogElement.classList.add(classes.show);
+          // if dialog is modal, disable scroll on page
+          document.documentElement.style.overflowY = 'hidden';
+        }
+      } else if (dialogElement.open) {
+        handleDialogClose();
+      }
+
+      return () => {
+        if (dialogElement.open) {
+          dialogElement.close();
+        }
+      };
+    }, [open, handleDialogClose, hasNativeDialog]);
+
+    const onDialogClick = (
+      event: ClickEvent<HTMLDialogElement> | ClickEvent<HTMLDivElement>,
+    ) => {
+      if (event.target === event.currentTarget && !preventClose) {
+        handleDialogClose();
+      }
+    };
+
+    return (
+      <>
+        {/* eslint-disable-next-line  jsx-a11y/no-noninteractive-element-interactions */}
+        <dialog
+          onClick={onDialogClick}
+          ref={applyMultipleRefs(ref, dialogRef)}
+          className={clsx(
+            className,
+            classes.dialog,
+            variant === 'immersive' ? classes.immersive : '',
```

Comment:
`clsx` filters out falsy values.
```suggestion
            variant === 'immersive' && classes.immersive,
```